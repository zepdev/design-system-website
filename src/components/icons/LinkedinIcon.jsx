import React from "react"
import PropTypes from "prop-types"

const LinkedinIcon = ({ className, width, height, fill, viewBox }) => (
  <svg
    width={width}
    height={height}
    xmlns="http://www.w3.org/2000/svg"
    viewBox={viewBox}
    aria-labelledby="title_linkedin"
    className={className}
  >
    <title id="title_linkedin">Linkedin</title>
    <path
      fill={fill}
      d="M21,21 L17,21 L17,14.25 C17,13.19 15.81,12.31 14.75,12.31 C13.69,12.31 13,13.19 13,14.25 L13,21 L9,21 L9,9 L13,9 L13,11 C13.66,9.93 15.36,9.24 16.5,9.24 C19,9.24 21,11.28 21,13.75 L21,21 L21,21 Z M7,21 L3,21 L3,9 L7,9 L7,21 L7,21 Z M5,3 C6.1045695,3 7,3.8954305 7,5 C7,6.1045695 6.1045695,7 5,7 C3.8954305,7 3,6.1045695 3,5 C3,3.8954305 3.8954305,3 5,3 Z"
    />
  </svg>
)

LinkedinIcon.propTypes = {
  width: PropTypes.string,
  height: PropTypes.string,
  viewBox: PropTypes.string,
  fill: PropTypes.string,
}

LinkedinIcon.defaultProps = {
  width: "32",
  height: "32",
  viewBox: "0 0 24 24",
  fill: "currentColor",
}

export default LinkedinIcon
