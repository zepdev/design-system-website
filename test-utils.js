import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@material-ui/core/styles'
import theme from './src/theme'

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
