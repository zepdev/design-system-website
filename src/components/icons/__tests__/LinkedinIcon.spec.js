import React from 'react'
import { render } from '../../../../test-utils'
import LinkedinIcon from '../LinkedinIcon'

describe('LinkedinIcon', () => {
  it('renders correctly', () => {
    const component = render(<LinkedinIcon />)
    expect(component).toMatchSnapshot()
  })
})
