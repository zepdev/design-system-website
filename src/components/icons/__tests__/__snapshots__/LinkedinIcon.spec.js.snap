// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkedinIcon renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        aria-labelledby="title"
        height="32"
        viewBox="0 0 24 24"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="title_linkedin"
        >
          Linkedin Icon
        </title>
        <path
          d="M21,21 L17,21 L17,14.25 C17,13.19 15.81,12.31 14.75,12.31 C13.69,12.31 13,13.19 13,14.25 L13,21 L9,21 L9,9 L13,9 L13,11 C13.66,9.93 15.36,9.24 16.5,9.24 C19,9.24 21,11.28 21,13.75 L21,21 L21,21 Z M7,21 L3,21 L3,9 L7,9 L7,21 L7,21 Z M5,3 C6.1045695,3 7,3.8954305 7,5 C7,6.1045695 6.1045695,7 5,7 C3.8954305,7 3,6.1045695 3,5 C3,3.8954305 3.8954305,3 5,3 Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      aria-labelledby="title"
      height="32"
      viewBox="0 0 24 24"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title
        id="title_linkedin"
      >
        Linkedin Icon
      </title>
      <path
        d="M21,21 L17,21 L17,14.25 C17,13.19 15.81,12.31 14.75,12.31 C13.69,12.31 13,13.19 13,14.25 L13,21 L9,21 L9,9 L13,9 L13,11 C13.66,9.93 15.36,9.24 16.5,9.24 C19,9.24 21,11.28 21,13.75 L21,21 L21,21 Z M7,21 L3,21 L3,9 L7,9 L7,21 L7,21 Z M5,3 C6.1045695,3 7,3.8954305 7,5 C7,6.1045695 6.1045695,7 5,7 C3.8954305,7 3,6.1045695 3,5 C3,3.8954305 3.8954305,3 5,3 Z"
        fill="currentColor"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
